scripts/package_building/build_artifacts.sh

```diff
@@ -0,0 +1,404 @@
+    fi
+    popd
+}
+
+function build_debian (){
+    #
+    # Building the kernel or daemons for deb based OSs
+    #
+    base_dir="$1"
+    source="$2"
+    build_state="$3"
+    thread_number="$4"
+    destination_path="$5"
+    
+    if [[ "$build_state" == "kernel" ]];then
+        pushd "$source"
+        make-kpkg --initrd kernel_image kernel_headers -j"${thread_number}"
+        popd
+    else
+        pushd "${base_dir}/daemons/hyperv-daemons"
+        debuild -us -uc
+        popd
+    fi
+    cp "${base_dir}/${build_state}/"*.deb "$destination_path"
+}
+
+function build_rhel {
+    #
+    # Building the kernel or daemons for rpm based OSs
+    #
+    base_dir="$1"
+    source="$2"
+    build_state="$3"
+    thread_number="$4"
+    destination_path="$5"
+    spec="$6"
+    
+    if [[ "$build_state" == "kernel" ]];then
+        pushd "$source"
+        make rpm -j"${thread_number}"
+        popd
+    else
+        pushd "${base_dir}/daemons/rpmbuild"
+        rpmbuild -ba "SPECS/$spec"
+        popd
+    fi
+    cp "${base_dir}/${build_state}/rpmbuild/RPMS/x86_64/"*.rpm "$destination_path"
+}
+
+function build_kernel (){
+    #
+    # Build the kernel
+    #
+    base_dir="$1"
+    source_path="$2"
+    os_family="$3"
+    download_method="$4"
+    destination_path="$5"
+    thread_number="$6"
+    build_state="kernel"
+    
+    prepare_env_"${os_family}" "$base_dir" "$build_state"
+    source="$(get_sources_${download_method} $base_dir $source_path)"
+    prepare_kernel_"${os_family}" "$source"
+    build_"${os_family}" "$base_dir" "$source" "$build_state" "$thread_number" "$destination_path"
+}
+
+function build_daemons (){
+    #
+    # Build the daemons
+    #
+    base_dir="$1"
+    source_path="$2"
+    os_family="$3"
+    download_method="$4"
+    debian_version="$5"
+    destination_path="$6"
+    dep_path="$7"
+    build_state="daemons"
+    
+    prepare_env_"${os_family}" "$base_dir" "$build_state"
+    prepare_daemons_"${os_family}" "$base_dir" "$source" "$debian_version" "$dep_path"
+    build_"${os_family}" "$base_dir" "$source" "$build_state" "$thread_number" "$destination_path" "$spec"
+}
+
+function clean_env_debian (){
+    #
+    # Removing sources and files required by the build process.
+    #
+    base_dir="$1"
+    
+    rm -rf "${BASE_DIR}/"*
+}
+
+function clean_env_rhel (){
+    #
+    # Removing sources and files required by the build process.
+    #
+    base_dir="$1"
+    
+    rm -rf "${BASE_DIR}/"*
+    rm -r "$HOME/.rpmmacros"
+}
+
+function main {
+    os_info="$(get_os_version)"
+    #Ex : os_info="os_VENDOR=Ubuntu os_RELEASE=16.04 os_UPDATE=4 os_PACKAGE=deb os_PACKAGE_MANAGER=apt os_CODENAME=xenial"
+    for info in $os_info;do
+        var_name="${info%=*}"
+        var_value="${info#*=}"
+        declare $var_name="$var_value"
+    done
+    
+    BASE_DIR="$(pwd)/temp_build"
+    DEP_PATH="$(pwd)/deps-lis/${os_PACKAGE}"
+    USE_CCACHE="False"
+    GIT_BRANCH="master"
+    CLEAN_ENV="False"
```

Comment:
make sure, if clean_env is set to false, to not delete the already built kernel sources, and just update the kernel source code.